code/missionui/redalert.cpp

```diff
@@ -734,6 +734,8 @@ void red_alert_store_wingman_status()
 	// store status for all existing ships
 	for ( so = GET_FIRST(&Ship_obj_list); so != END_OF_LIST(&Ship_obj_list); so = GET_NEXT(so) ) {
 		ship_objp = &Objects[so->objnum];
+		if (ship_objp->flags[Object::Object_Flags::Should_be_dead])
+			continue;
```

Comment:
So, this one is a little tricky.  I think we need to check that vanished ships will be added to the Ships_exited container so that we don't accidentally leave out a ship.

Reply:
I just looked at this and vanished ships are intentionally not added to Ships_exited.  What are you referring to by "accidentally leave out"?

Reply:
So let's say the vanished ship was in the starting wings.  I'm not sure what the result would be if red alert processing then skipped that ship here because it was previously vanished or has the `Should_be_dead` flag.  Would that ship be missing in hud gauge on the next mission? Or would something else possibly worse happen.

Additionally, if vanished ships are never added to `Ships_exited` then it could be that vanished ships in starting wings are not handled correctly at all.  This is a very specific case.  If vanished ships are not handled correctly, anyway, then this change will not really change behavior except to also exclude ships that were vanished in the same frame that red alert is triggered. 

At some point someone may want to make sure that vanished ships in starting wings are handled correctly in red alert.

Reply:
But this change seems to reinforce the current behavior, at least, whether it's buggy or works.  So I don't think it's an issue here.  I just really think someone should make sure the two features work together as intended.

Reply:
If a ship in the starting wings is vanished, it would be treated as if it wasn't included in the red-alert at all; that is, if the same ship appears in the next mission, it would be a "brand new" ship.